### PR TITLE
Add docstring for Tar module

### PR DESCRIPTION
**Pull request by lgoettgens**
Handles the `Tar` part of https://github.com/JuliaLang/julia/issues/52725.

cc @stevengj

### CODE REVIEW

File: docs/src/index.md
Comment (stevengj) [inline docs/src/index.md:5]: Maybe something like:
```suggestion
The `Tar` module provides a simple interface for handling tar archives, including creation of
archives, extraction of selected files from an archive, and access to metadata.
```

File: docs/src/index.md
Comment (stevengj) [inline docs/src/index.md:9]: I notice that none of these functions are currently exported.    In Julia 1.11, they should be declared public, e.g. something like:
```jl
@static if VERSION >= v"1.11"
    public create, extract, list, rewrite, tree_hash, Header
end
```
since this is apparently the documented API.

File: test/runtests.jl
Comment (stevengj) [inline test/runtests.jl:1136]: You need to run this test only in recent Julia versions:
```suggestion
if isdefined(Docs, :undocumented_names) # new in Julia 1.11
    @testset "Docstrings" begin
        @test Docs.undocumented_names(Tar) == []
    end
end
```

File: src/Tar.jl
Comment (lgoettgens) [inline src/Tar.jl:2]:
```suggestion
The `Tar` module provides a simple interface for handling tar archives, including creation of
archives, extraction of selected files from an archive, and access to metadata.
```

File: docs/src/index.md
Comment (lgoettgens) [reply docs/src/index.md:10]: I think this is out of scope of this PR. I'll reference it in https://github.com/JuliaLang/julia/issues/51335 instead to not forget.

File: test/runtests.jl
Comment (lgoettgens) [reply test/runtests.jl]: Maybe you can adapt the second line of https://github.com/JuliaLang/julia/issues/52725#issue-2064789767, where I copied the test from?

**Review by stevengj — APPROVED** [review]

File: docs/src/index.md
Comment (LilithHafner) [reply docs/src/index.md:10]: > I'm not sure if there is a way to avoid `eval(Meta.parse(...))` since the public keyword does not parse in earlier Julia versions?

The only good way to avoid that that I know of is to use Compat.jl's `@compat public foo, bar`, though this is a fine use of `eval(Meta.parse(....))` imo.

Comment (vtjnash) [reply docs/src/index.md:10]: Or make a small macro here `macro public(v); @assert isexpr(v, :tuple); VERSION ? esc(Expr(:public, v.args...)); nothing; end` so that it just needs to add an `@` to make it syntactically valid here

Comment (LilithHafner) [reply docs/src/index.md:10]: That macro (with some bug-fixes) is here https://github.com/JuliaLang/Compat.jl/pull/805/files/0c90fae7a869a5f526a4cb13620ea077fc30456e, though it was renamed to `@compat public`.

Comment (lgoettgens) [reply docs/src/index.md:10]: Can we get this PR here merged? The discussion about declaring public symbols can be moved to #174 and/or #175. There I already implemented two different approaches

**Review by LilithHafner — APPROVED** [review]
LGTM

@stevengj, is there anything you're waiting for before merge

**Pull request merged**
